pkg/tecdsa/dkg/protocol_test.go

```diff
@@ -1313,144 +1313,6 @@ func TestVerifyDKGResultSignatures(t *testing.T) {
 				3:                         []byte("sign 3"),
 			},
 		},
-		"multiple messages from other member with different signatures for " +
```

Comment:
My understanding of the reasoning behind removing these unit tests is that all these cases do not matter because only the first message is accepted. On the other hand, there is no guarantee the deduplication code is called from `pkg/tescdsa/dkg/states.go` so maybe it would make sense to leave one or two of these tests that would cover that the deduplication works properly? Right now, looking at just the unit tests, it is unclear what happens if the same member sends two messages with correct or incorrect signature.

Reply:
We need to remove those tests because we made a change in messages validation. So far, each `protocol.go` function was responsible for duplication of its own state messages, hence the test cases. Now, we put the deduplication on the higher abstraction level and this is done in `states.go` so `protocol.go` functions work with an assumption that the messages received as input are correctly deduplicated, hence we need to remove those cases.

Reply:
I see. I think we should implement unit tests, similar to the ones in `pkg/beacon/dkg/result/states_test.go` where we instantiate member and state manually, to confirm the state's `Initiate` uses `receivedMessages` to filter out duplicates. Alternatively, at least cover the `receivedMessages` with a unit test.

Reply:
Added unit tests for `receivedMessages`, just like we have for signing: https://github.com/keep-network/keep-core/pull/3392/commits/846f158d4c91ab2947bdb831d42ba35ce7405cf8